tooling/noir_js_types/lib/types.ts

```diff
@@ -0,0 +1,18 @@
+export interface Backend {
+  // Generate an outer proof. This is the proof for the circuit which will verify
+  // inner proofs and or can be seen as the proof created for regular circuits.
+  generateProof(decompressedWitness: Uint8Array, optimizeForVerifyInCircuit?: boolean): Promise<Uint8Array>;
+
+  // Generates a child proof. Child Proof will be verified in another Circuit.
+  generateChildProof(decompressedWitness: Uint8Array): Promise<Uint8Array>;
```

Comment:
Noting that this has changed from https://github.com/noir-lang/noir/pull/2856

Reply:
Did we end up with this naming? I kind of like @Savio-Sou suggestion of "intermediate proof" and "final proof" as it sound more descriptive

Reply:
Does that make sense in case where you are not doing recursion?

Reply:
> Does that make sense in case where you are not doing recursion?

The user would stick with `generateFinalProof` then. Not the cleanest but not the most unintuitive (while more intuitive than `generateProof` for recursion users, which could be a big group of devs as recursion would be needed for any fairly complex programs running in browsers.)